packages/lesswrong/server/repos/PostsRepo.ts

```diff
@@ -776,6 +784,69 @@ class PostsRepo extends AbstractRepo<"Posts"> {
     `, 
     [userId, limit]);
   }
+  
+  async getPostsAndCommentsFromSubscriptions(userId: string, numDays: number): Promise<Array<PostAndCommentsResultRow >> {
+    return await this.getRawDb().manyOrNone<PostAndCommentsResultRow>(`
+      WITH user_subscriptions AS (
+      SELECT DISTINCT "displayName", type, "documentId" AS "userId"
+      FROM "Subscriptions" s
+               LEFT JOIN "Users" u ON u._id = s."documentId"
+      WHERE state = 'subscribed'
+        AND s.deleted IS NOT TRUE
+        AND "collectionName" = 'Users'
+        AND "type" = 'newActivityForFeed'
+        AND "userId" = $1
+      ),
+      posts_by_subscribees AS (
+           SELECT
+              p._id AS "postId",
+              "postedAt",
+              TRUE as "subscribedPosts" --,
+           FROM "Posts" p
+                    JOIN user_subscriptions us USING ("userId")
+           WHERE p."postedAt" > CURRENT_TIMESTAMP - INTERVAL $2
+           ORDER BY p."postedAt" DESC
+       ),
+      posts_with_comments_from_subscribees AS (
+          SELECT
+             "postId",
+             ARRAY_AGG(c._id ORDER BY c."postedAt" DESC) AS "commentIds",
+             MAX(c."postedAt") AS last_updated,
+            TRUE as "subscribedComments"
+          FROM "Comments" c
+               JOIN user_subscriptions us USING ("userId")
+               JOIN "Revisions" r ON c.contents_latest = r._id
+          WHERE c."postedAt" > CURRENT_TIMESTAMP - INTERVAL $2
+          AND (c."baseScore" >= 5 OR r."wordCount" >= 200)
+          AND c.deleted IS NOT TRUE
+          AND c."authorIsUnreviewed" IS NOT TRUE
+          AND c.retracted IS NOT TRUE
+          GROUP BY "postId"
+      ),
+      combined AS (
+          SELECT
+              *
+          FROM
+              posts_by_subscribees
+          FULL JOIN posts_with_comments_from_subscribees USING ("postId")
+      )
+      SELECT combined.*
+      FROM combined
+      JOIN "Posts" p ON combined."postId" = p._id
+      WHERE
+        p.draft IS NOT TRUE
+        AND p.status = 2
+        AND p.rejected IS NOT TRUE
+        AND p."authorIsUnreviewed" IS NOT TRUE
+        AND p."hiddenRelatedQuestion" IS NOT TRUE
+        AND p.unlisted IS NOT TRUE
+        AND p."isFuture" IS NOT TRUE
```

Comment:
I have a vague memory of running into some performance issues when writing a query similar to this and using `IS NOT TRUE` for all of these conditions, instead of `IS FALSE` (which is weird, but 🤷).  But if you've benchmarked it (i.e. `EXPLAIN ANALYZE`) and it's <100ms, then probably fine

Reply:
(The thing that fixed it was making the one condition that was causing a problem match the way that condition got translated from the default posts view.)